### DISCUSSION TITLE
Can this be installed on arch linux?

### DISCUSSION THREAD

**Discussion by HenrikPassmark** [PromtEngineer/localGPT#116] [Q&A]
Can this be installed on arch linux?

**Comment by KonradHoeffner** [comment]
Without CUDA and cuBLAS using CPU only: Yes. I got some system freezes with XMP active but that is probably my fault for mixing RAM from different vendors (hasn't been a problem in anything else).

With CUDA and cuBLAS:  I have tried for several hours but failed. This was mostly due to dependency conflicts with CUDA and gcc. Arch Linux provides CUDA 12 which is not compatible with localGPT. I tried downgrading a bunch of packages but this would ultimately break the system so I stopped.

However to get around this I added a Dockerfile in <https://github.com/PromtEngineer/localGPT/pull/279>.
If you have Docker installed, it should work (can't test due to not enough GPU RAM) so it could be great if you can report if it works for you.

P.S.: Here are some more notes:

Arch Linux has CUDA 12 which is not supported by PyTorch.
I tried installing CUDA 11.7 from the Arch Linux package archives but that depends on gcc 11 which depends on gcc-libs 11 and downgrading that breaks the system.
Also tried installing gcc 11 from AUR but that was still compiling after an hour and its unclear if CUDA will actually choose the correct one if both are installed.
I could install gcc 11 in the local Conda environment but CUDA 12 wasn't using that.
cuda-11.0 and cudnn8-cuda11.0 from AUR still fails at `pip install .` in the AutoGPTQ directory version 0.2.2 with `RuntimeError: The current installed version of g++ (13.1.1) is greater than the maximum required version by CUDA 11.0.
Please make sure to use an adequate version of g++ (>=5.0.0, <10.0).`.
Manually changing the maximum g++ version in `cpp_extension.py` also produces an error, it seems to be there for a reason.
